src/scheduler.cc

```diff
@@ -55,6 +65,7 @@ namespace detail {
 	void scheduler::shutdown() {
 		abstract_scheduler::shutdown();
 		if(worker_thread.joinable()) { worker_thread.join(); }
+		assert(available_events.empty() && in_flight_count == 0);
```

Comment:
This is missing a lock on `events_mutex` (can potentially race with `notify()`). It should be enough to take it in a `#ifndef NDEBUG' though.

Reply:
Right, although this is moot now, since the refactoring into `abstract_scheduler` doesn't allow me to access these members anyway. I've removed the assertion.